app/models/grda_warehouse/hud/client.rb

```diff
@@ -1785,6 +1785,22 @@ def last_projects_served_by(include_confidential_names: false)
       end.uniq.sort
     end
 
+    def last_intentional_contacts_ignore_visibile_status(include_confidential_names: false, skip_confidential_projects: false, include_dates: false)
+      contacts = processed_service_history&.last_intentional_contacts
+      return [] unless contacts.present?
+
+      contacts = JSON.parse(contacts)
+
+      contacts.sort_by { |c| c['date']&.to_date || 5.years.ago }.reverse.map do |contact|
+        project = GrdaWarehouse::Hud::Project.find(contact['project_id'])
```

Comment:
This is going to introduce an N+1 (and sometimes we're loading data for 60k+ clients when pushing them to CAS, I don't have a great solution right now, other than fetching all the projects and passing in a hash of them.  Open to other ideas.

Reply:
I've pushed a new commit to pull the projects ahead of time to clean the N+1